### ISSUE TITLE
IO support for NB models 

### ISSUE THREAD

**Issue by OmriTreidel** [dfdx/NaiveBayes.jl#10]
It would be great to have the option of exporting a model once the training is finished and reuse that model many times by importing it. Something like

``` julia
m = Hybrid(labels, 2, 3)
fit(m, training_features_c, training_features_d)
export(m, model_file.hdf5) 
```

and 3 days later

``` julia
m = import(model_file.hdf5)
y = predict(m, features_c, features_d)
```

@dfdx ,@JoshChristie any thoughts? What file format/s should we support? I'm keen on HDF5 since it save the trouble of parsing.

It also makes a lot of sense to store names for the features in the NB mode.

**Comment by JoshChristie** [comment]
Sounds good. Overloading import won't work though.
I like HDF5 or checkout https://github.com/JuliaIO/JLD.jl for automatic saving type info.

**Comment by OmriTreidel** [comment]
JLD seems like the easiest option that would work for all of the models. What are the downsides?

**Comment by dfdx** [comment]
I [used HDF5](https://github.com/dfdx/Boltzmann.jl/blob/master/src/io.jl) for Boltzmann.jl and am pretty happy with it (although now I think API in that file could be improved). JLD relies on HDF5 and is more Julia-specific (read as "not usable from other languages"), but probably it's not a an issue, at least for now.

The most important issue to keep in mind is that models evolve and thus serialization format should try to be backward compatible. In particular, we should care about:
- compatibility between Julia versions
- compatibility between versions of the model itself

The easiest way to achieve it should be to save only parameters (e.g. numbers, vectors and matrices), but not the model itself or anything closely related to the implementation. It would be pretty unpleasant to change a minor detail on code and find out that pretrained model doesn't load anymore.

**Comment by OmriTreidel** [comment]
I completely agree with the choice of HDF5 and that we should simply save that data (parameters and distribution). But there is one more important thing: the way the features were computed. Imagine that you compute a features X with some parameters Ax, train your model, in tomorrow you are trying to classify (using the same model) features X that was computed with a different parameter Bx. This metadata is crucial and in the case of HDF5 is easy to store it as the fields names.

**Comment by dfdx** [comment]
Closed by #11
